libp2p/protocols/connectivity/autonatv2/utils.nim

```diff
@@ -0,0 +1,110 @@
+{.push raises: [].}
+
+import results, strutils
+import chronos
+import
+  ../../protocol,
+  ../../../switch,
+  ../../../multiaddress,
+  ../../../multicodec,
+  ../../../peerid,
+  ../../../protobuf/minprotobuf,
+  ../autonat/service,
+  ./types
+
+proc ipAddrMatches*(observedIPAddr: MultiAddress, reqAddrs: seq[MultiAddress]): bool =
+  ## Checks if at least one of the requests' addresses
+  ## has the same IP address of observedIPAddr
+
+  let observedIPAddr = observedIPAddr[0].valueOr:
+    return false
+
+  for addr in reqAddrs:
+    addr[0].withValue(ipAddr):
+      if ipAddr == observedIPAddr:
+        return true
+  return false
+
+proc ipSupport*(s: Switch): (bool, bool) =
+  ## Returns ipv4 and ipv6 support status of a switch
+
+  var ipv4 = false
+  var ipv6 = false
+
+  for addrs in s.peerInfo.listenAddrs:
+    addrs[0].withValue(addrIp):
+      if IP4.match(addrIp):
+        ipv4 = true
+      elif IP6.match(addrIp):
+        ipv6 = true
+
+  (ipv4, ipv6)
+
+proc isPrivateIP*(ip: string): bool {.raises: [ValueError].} =
```

Comment:
AutoTLS has a `isPublic`. Could we maybe put both that function and this one in some utils file?